martin/src/srv/tiles.rs

```diff
@@ -100,24 +105,35 @@ impl<'a> DynTileSource<'a> {
             query_str,
             query_obj,
             accept_enc,
+            if_none_match,
             preferred_enc,
             cache,
         })
     }
 
     pub async fn get_http_response(&self, xyz: TileCoord) -> ActixResult<HttpResponse> {
         let tile = self.get_tile_content(xyz).await?;
+        let hash = xxhash_rust::xxh3::xxh3_128(&tile.data);
+        let etag = EntityTag::new_strong(hash.to_string());
 
-        Ok(if tile.data.is_empty() {
-            HttpResponse::NoContent().finish()
-        } else {
-            let mut response = HttpResponse::Ok();
-            response.content_type(tile.info.format.content_type());
-            if let Some(val) = tile.info.encoding.content_encoding() {
-                response.insert_header((CONTENT_ENCODING, val));
+        if tile.data.is_empty() {
```

Comment:
Maybe check is_empty before hash calculation?

```suggestion
        if tile.data.is_empty() {
            return Ok(HttpResponse::NoContent().finish());
        }
        let hash = xxhash_rust::xxh3::xxh3_128(&tile.data);
        let etag = EntityTag::new_strong(hash.to_string());
```